### PR TITLE
stop deleting clients when no prekeys exist for these clients

### PR DESCRIPTION
**Pull request by stefanwire**
https://wearezeta.atlassian.net/browse/FS-966

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by jschaul — COMMENTED** [review]
I'm not convinced the approach of pushing code to production then looking at logs is fruitful in the backend's case; the feedback loop is too long and the risk of damage done of leaving *more* state inconsistency for more users is high. If this state can be achieved with current code, then it's possible to write an integration test showing clients without prekeys existing either in brig, or galley, or both. Then the production code could be changed.

The above is my gut feeling; I've not been involved with this issue before and don't have prior context; but it sounds like a bug regarding clients was introduced accidentally at some point in the recent past; Perhaps a review of code changes here would be more fruitful.

File: services/brig/src/Brig/API/Client.hs [hunk 1]
```diff
@@ -404,7 +409,8 @@ noPrekeys u c = do
     field "user" (toByteString u)
       ~~ field "client" (toByteString c)
       ~~ msg (val "No prekey found. Ensuring client does not exist.")
-  Intra.rmClient u c
+  -- Intra.rmClient u c
```
Comment (elland) [inline services/brig/src/Brig/API/Client.hs:412]: I assume at some point we need to actually delete the clients correctly everywhere or is that not the case? Also would that require some script to prune the required Casandras?

**Comment by stefanwire** [comment]
The discussion continues in https://wearezeta.atlassian.net/browse/FS-966.

**Review by jschaul — CHANGES_REQUESTED** [review]
"requesting changes" here to block this PR from being merged until the discussion in FS-966 has lead to a consensus.

**Comment by smatting** [comment]
Closing this. See details in https://wearezeta.atlassian.net/browse/FS-966

**Pull request closed without merging**